Refactor Update Checker

duk3luk3 opened the pull request:
`GithubUpdateChecker` -> `UpdateChecker`

Collects Github stable release, github prerelease, and server update
info

For #671 

- [x] PR branch should be named `issuenum`-`fix`/`feature`/`cleanup`-`description`
- [x] Code is split into logical commits
- [ ] Code has tests
- [x] Final commit includes "Fixes #issue" in commit message

When all builds pass and a maintainer is happy with your PR, the "ready" label will be applied. Please complete these tasks then:

- [x] Rebase onto develop
- [x] Add changelog entry
- [ ] Remove this entire template section

Wesmania commented on src/client/updater.py:
```diff
@@ -15,33 +15,64 @@
+
+    def _parse_release(self, release_info):
+            try:
+                js = json.loads(release_info))[0]
```
s@))@)@

muellni commented:
Looks good. 
I'm having a small problem on my machine: The `update_settings.ui` dialog can be resized such that the `lblExplanation` text is truncated. IDK if that happens on Windows too.

muellni commented:
https://github.com/duk3luk3/client/pull/2 should fix this (and renames the dialog to "Update Settings")

muellni commented:
and here's another little fix: https://github.com/duk3luk3/client/pull/3

duk3luk3 committed:
Refactor Update Checker

`GithubUpdateChecker` -> `UpdateChecker`

Collects Github stable release, github prerelease, and server update
info and presents a clean, simple dialog to the user.

Also allows the user to select which version branch they want to get
release updates for.

Fixes #671
